keyboards/system76/launch_1/launch_1.c

```diff
@@ -0,0 +1,205 @@
+    eeconfig_disable();
+}
+
+void system76_ec_rgb_eeprom(bool write);
+void system76_ec_rgb_layer(layer_state_t layer_state);
+void system76_ec_unlock(void);
+bool system76_ec_is_unlocked(void);
+
+rgb_config_t layer_rgb[DYNAMIC_KEYMAP_LAYER_COUNT];
+
+void matrix_init_kb(void) {
+    usb_mux_init();
+
+    bootmagic_lite();
+    if (!eeprom_is_valid()) {
+        dynamic_keymap_reset();
+        dynamic_keymap_macro_reset();
+        system76_ec_rgb_eeprom(true);
+        eeprom_set_valid(true);
+    } else {
+        system76_ec_rgb_eeprom(false);
+    }
+
+    system76_ec_rgb_layer(layer_state);
+}
+
+void matrix_scan_kb(void) {
+    usb_mux_event();
+
+    matrix_scan_user();
+}
+
+#define LEVEL(value) (uint8_t)(((uint16_t)value) * ((uint16_t)RGB_MATRIX_MAXIMUM_BRIGHTNESS) / ((uint16_t)255))
+
+// clang-format off
+static const uint8_t levels[] = {
+    LEVEL(48),
+    LEVEL(72),
+    LEVEL(96),
+    LEVEL(144),
+    LEVEL(192),
+    LEVEL(255)
+};
+// clang-format on
+
+static uint8_t toggle_level = RGB_MATRIX_MAXIMUM_BRIGHTNESS;
+extern bool    input_disabled;
+
+static void set_value_all_layers(uint8_t value) {
+    if (!system76_ec_is_unlocked()) {
+        for (int8_t layer = 0; layer < DYNAMIC_KEYMAP_LAYER_COUNT; layer++) {
+            layer_rgb[layer].hsv.v = value;
+        }
+        system76_ec_rgb_layer(layer_state);
+    }
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (input_disabled) { return false; }
+
+    switch (keycode) {
+        case RESET:
+            if (record->event.pressed) {
+                system76_ec_unlock();
+            }
+            return false;
+        case RGB_VAD:
+            if (record->event.pressed) {
+                uint8_t level = rgb_matrix_config.hsv.v;
+                for (int i = sizeof(levels) - 1; i >= 0; i--) {
+                    if (levels[i] < level) {
+                        level = levels[i];
+                        break;
+                    }
+                }
+                set_value_all_layers(level);
+            }
+            return false;
+        case RGB_VAI:
+            if (record->event.pressed) {
+                uint8_t level = rgb_matrix_config.hsv.v;
+                for (int i = 0; i < sizeof(levels); i++) {
+                    if (levels[i] > level) {
+                        level = levels[i];
+                        break;
+                    }
+                }
+                set_value_all_layers(level);
+            }
+            return false;
+        case RGB_TOG:
+            if (record->event.pressed) {
+                uint8_t level = 0;
+                if (rgb_matrix_config.hsv.v == 0) {
+                    level = toggle_level;
+                } else {
+                    toggle_level = rgb_matrix_config.hsv.v;
+                }
+                set_value_all_layers(level);
+            }
+            return false;
+    }
+
+    return process_record_user(keycode, record);
```

Comment:
Ideally `process_record_user()` should be called before the rest of code in `process_record_kb()`, so that it would be possible to override the keyboard-level behavior from the keymap:
```c
    if (!process_record_user(keycode, record)) {
        return false;
    }
    switch (keycode) {
        // ... the rest of process_record_kb() code ...
    }
```
Although the `input_disabled` handling in this particular case might still need to go before `process_record_user()` (but there would still be some code that runs before that — in particular, the combo code may invoke callbacks which won't get disabled).

Reply:
From browsing through the other System76 repositories, `input_disabled` feature appears to be used for testing. Moving as suggested.